src/agent/instance_communicator/src/instance_communicator_win.cpp

```diff
@@ -0,0 +1,96 @@
+#include <instance_communicator.hpp>
+
+#include <logger.hpp>
+
+#include <windows.h>
+
+#include <iostream>
+
+namespace instance_communicator
+{
+    const std::string pipeName = "\\\\.\\pipe\\agent-pipe";
+
+    boost::asio::awaitable<void> AsyncConnectNamedPipe(boost::asio::windows::stream_handle& pipe,
+                                                       boost::system::error_code& ec)
+    {
+        HANDLE native = pipe.native_handle();
+        BOOL result = ConnectNamedPipe(native, nullptr);
+
+        if (!result && GetLastError() != ERROR_PIPE_CONNECTED)
+        {
+            ec = boost::system::error_code(GetLastError(), boost::system::system_category());
+            co_return;
+        }
+
+        ec = {};
+        co_return;
+    }
+
+    boost::asio::awaitable<void> InstanceCommunicator::Listen(
+        boost::asio::io_context& ioContext) // NOLINT (cppcoreguidelines-avoid-reference-coroutine-parameters)
+    {
+        // LogWarn(L"Setting InstanceCommunicator to listen on named pipe {}", pipeName.c_str());
+
+        while (m_keepRunning.load())
+        {
+            try
+            {
+                // Create named pipe
+                HANDLE pipeHandle = CreateNamedPipe(pipeName.c_str(),
```

Comment:
Is it necessary to create a named pipe in each iteration? Can't we create this just once before the main loop (something similar to the acceptor)?